### PR TITLE
woop: uses combobox to navigate countries

### PR DESCRIPTION
**Pull request by Rolilink**
#### Changes proposed in this Pull Request
* Updating Country/Region dropdown to use ComboboxControl

Screen Shot 2022-01-10 at 2 25 40 PM

#### Testing instructions
* Apply D72720-code and sandbox the API.
* Navigate to `http://calypso.localhost:3000/start/woocommerce-install/store-address?site=<siteId>`
* Select a country (and region, if available) in the country dropdown
* Fill the rest and click continue
* With D72639-code it should send an http request to settings with `woocommerce_default_country` being the code of the selected country.

Related to #
* https://github.com/Automattic/wp-calypso/issues/59833

### CODE REVIEW

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (lsl) [inline client/signup/steps/woocommerce-install/step-store-address/index.tsx]:
```suggestion
						label={ __( 'Country / Region', 'woocommerce-admin' ) }
```

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (sirbrillig) [inline client/signup/steps/woocommerce-install/step-store-address/index.tsx]: The `name` prop does not exist on `ComboboxControl`.

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (sirbrillig) [inline client/signup/steps/woocommerce-install/step-store-address/index.tsx]: The `onChange` callback for this function can sometimes be called with a `null` value, so you'll need to handle that here. Perhaps just with an early return?

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (sirbrillig) [inline client/signup/steps/woocommerce-install/step-store-address/index.tsx:76]: This component does not yet have types in the package definition for `@wordpress/components` which is causing your type errors. I've added them in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58106. Once that merges and the calypso types are updated, this should pass the unit tests. The Calypso team can cause it to pass sooner if you ask!

Comment (lsl) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx:76]: Thanks, we might do that. cc @p-jackson since we're midchat about this :laughing:

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (lsl) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: Null looks like they happen when you click the x reset button, just set this to empty string (c601fea), we'll be validating these values shortly.

File: client/signup/steps/woocommerce-install/step-store-address/style.scss
Comment (lsl) [inline client/signup/steps/woocommerce-install/step-store-address/style.scss]:
```suggestion
	width: unset;
```

**Comment by lsl** [comment]
Some kind of encoding happening on the output:

Screenshot 2022-01-11 at 13-52-25 Add WooCommerce to your site — WordPress com

edit: this is coming from [translations](https://translate.wordpress.com/projects/wpcom/fr/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=296830&filters%5Btranslation_id%5D=9221215), french and variants seems to have a lot of these other languages seem to be ok. Lets ignore for now, not something we need to fix.

@DustyReagan `html_entity_decode` in D72720-code will "fix" this but I'm not sure that's the right approach, pretty sure these should be rejected at the translation layer. Pinged #i18n here: p1641880808023800-slack-C02AED43D

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (lsl) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: d0ffd5a

File: client/signup/steps/woocommerce-install/step-store-address/index.tsx
Comment (sirbrillig) [inline client/signup/steps/woocommerce-install/step-store-address/index.tsx]: I'm not sure why exactly, but TS believes `get( WOOCOMMERCE_DEFAULT_COUNTRY )` can return a `number`, which causes a type error since the `value` prop must be a string. Do you believe this should be a string? If so, can you cast it to a String here?

Comment (lsl) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: Is it stemming from [this](https://github.com/Automattic/wp-calypso/pull/59505/files#diff-24648f1a5849a32e3be548a0a3991dc3cafc59a3574d553d9dd75de8225aed29R40)? `as string` should be fine here right? I don't think there is any cases where a non string can enter this setting.

Comment (lsl) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: @retrofox is `blog_public` required in this hook? Can't see where/how it's used (still catching up after the break) - might be faster to just remove the def.

Comment (allilevine) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: @sirbrillig What's the best way to catch errors like these going forward? I wasn't seeing a type error there. 🤔 

I cast it to a string here: https://github.com/Automattic/wp-calypso/pull/59876/commits/ab4b8ac7918a2eb27eb5d2a15405b87bd56c1749

> Can't see where/how it's used (still catching up after the break) - might be faster to just remove the def.

@lsl I'm not seeing this either but it seems like `get` should be able to return different types, so I'm hesitant to limit it to strings.

Comment (sirbrillig) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: Good question! Sorry I didn't explain that earlier. You'd see the type errors if the types for `ComboboxControl` were available, but we're going to have to wait for the changes I merged there to be released and imported into calypso (and then rebased into this branch). Here's an example of when this was done recently: https://github.com/Automattic/wp-calypso/pull/59779

The reason I see them is that I manually added the types to my `node_modules` folder so I could check that they worked correctly, and that's why I was mentioning the errors to you here. This sort of thing is really weird and probably won't happen often.

Comment (sirbrillig) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: > but we're going to have to wait for the changes I merged there to be released and imported into calypso (and then rebased into this branch)

Since it's not clear to me how long this process takes, you may want to ping the calypso team and ask them to mark these unit tests as passing (assuming the missing `ComboboxControl` is still the only new error). I'd do it but I don't have that power. 😅

Comment (allilevine) [reply client/signup/steps/woocommerce-install/step-store-address/index.tsx]: > The reason I see them is that I manually added the types to my node_modules folder so I could check that they worked correctly, and that's why I was mentioning the errors to you here. 

Ahhh! Got it, thanks!

**Review by obenland — APPROVED** [review]
Looking good to me. 

Should we wait until https://github.com/Automattic/jetpack/pull/22248 made it to dotcom before merging it?

**Comment by obenland** [comment]
This should be good to go now. 

@Automattic/team-calypso could we ask you to mark the unit tests as passing, [as suggested here](https://github.com/Automattic/wp-calypso/pull/59876#discussion_r782342538)?

**Comment by tyxla** [comment]
This was pending on #60002 essentially, but I've marked it as green now so you're unblocked to merge. Once #60002 lands, the related TS error that is being introduced here should get resolved.

**Pull request merged**
